nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -195,49 +445,34 @@ public void deleteNetworkElement(NetworkElement serviceFunctionChain) throws Exc
 
     @Override
     public List<NetworkElement> getNetworkElements(NetworkElement serviceFunctionChain) throws Exception {
-        this.utils.throwExceptionIfNullElementAndId(serviceFunctionChain, "Service Function Chain Id");
+        checkArgument(serviceFunctionChain != null && serviceFunctionChain.getElementId() != null,
+                      "null passed for %s !", "Service Function Chain Id");
 
-        PortChain portChain = this.osClient.sfc().portchains().get(serviceFunctionChain.getElementId());
-        this.utils.throwExceptionIfCannotFindById(portChain, "Service Function Chain", serviceFunctionChain.getElementId());
+        PortChain portChain = this.osCalls.getPortChain(serviceFunctionChain.getElementId());
+
+        checkArgument(portChain != null && portChain.getId() != null,
+                      "Cannot find %s by id: %s!", "Service Function Chain", serviceFunctionChain.getElementId());
 
         if (portChain.getPortPairGroups() == null) {
             return emptyList();
         }
 
-        ServiceFunctionChainEntity sfcFound = new ServiceFunctionChainEntity(portChain.getId());
         ArrayList<PortPairGroupEntity> portPairGroupEntities = new ArrayList<>();
 
         for (String portPairGroupId : portChain.getPortPairGroups()) {
-            PortPairGroup portPairGroup = this.osClient.sfc().portpairgroups().get(portPairGroupId);
+            PortPairGroup portPairGroup = this.osCalls.getPortPairGroup(portPairGroupId);
 
             if (portPairGroup == null) {
```

Comment:
seems like redundant check. We just got the PPG ID's from SFC, why would the ppg be null here?